pandas/plotting/_matplotlib/tools.py

```diff
@@ -1,16 +1,22 @@
 # being a bit too dynamic
 from math import ceil
+from typing import TYPE_CHECKING, Tuple
 import warnings
 
 import matplotlib.table
 import matplotlib.ticker as ticker
 import numpy as np
 
+from pandas._typing import FrameOrSeries
+
 from pandas.core.dtypes.common import is_list_like
 from pandas.core.dtypes.generic import ABCDataFrame, ABCIndexClass, ABCSeries
 
 from pandas.plotting._matplotlib import compat
 
+if TYPE_CHECKING:
+    from matplotlib.table import Table
```

Comment:
Revealed type is 'Any'

not sure what's best here. I imagine that once we get np.ndarray imports sorted, there'll be a lot of mypy errors to fix. I wonder whether we want to continue adding types that currently resolve to Any or leave untyped.

Reply:
ill follow your lead on this.

there are a lot of places in pd.plotting where we have a `ax` arg that would be nice to disambiguate `pd.Index` vs `matplotlib.axes.Axes` vs `matplotlib.axis.Axis` (i think most are `Axes`)

Reply:
>  once we get np.ndarray imports sorted

is there cause for optimism this may be soon-ish?

Reply:
I'm not sure that we have got to any technical blocks yet since not much effort in this direction. Personally, I wanted to get the check_untyped_defs sorted first. (was a phase 0 in a plan) that imo comes before all other work. but in open source can't control the effort and therefore some have a preference for typing the public api. My interest in typing has always been for the internal consistency and quality of the pandas codebase itself. 

But with hindsight, getting typing PRs through was so painfull that any typing additions is a bonus.